Addressing some linter errors

sampathvic opened the pull request:
Summary: Addressing the linter errors reported in the changed files.

Test Plan:
```
buck test mode/opt deeplearning/fbgemm:QuantUtilsTest
```
https://www.internalfb.com/intern/testinfra/testrun/11821949118528688

```
buck test mode/opt caffe2/torch/fb/model_transform/splitting/tests:split_dispatcher_test
```
https://www.internalfb.com/intern/testinfra/testrun/7881299627525465

Rollback Plan:

Differential Revision: D78352311




cc @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @jerryzh168

cyyever commented on aten/src/ATen/native/quantized/cpu/fbgemm_utils.cpp:
```diff
@@ -112,24 +110,26 @@ fbgemm::conv_param_t<kSpatialDim> MakeFbgemmConvParam(
   std::array<int, kSpatialDim> kernels_{};
   std::array<int, kSpatialDim> strides_{};
-  std::array<int, kSpatialDim * 2> pads_{};
+  std::array<int, static_cast<std::size_t>(static_cast<long>(kSpatialDim) * 2)> pads_{};
```
This is computed at compile time, what was the warning?

sampathvic replied:
@cyyever Thanks for checking! This one: https://fburl.com/rav2373n

cyyever replied:
Is `kSpatialDim * 2ull` more simple?

sampathvic replied:
Yes.

cyyever commented on aten/src/ATen/native/quantized/cpu/fbgemm_utils.cpp:
```diff
@@ -83,10 +83,8 @@ void CopyICFirst3dTensorToChannelsLast3dTensor(
-        // NOLINTNEXTLINE(cppcoreguidelines-narrowing-conversions,bugprone-narrowing-conversions)
-        int oc = i % OC_G;
+        int g = static_cast<int>(i / OC_G);
+        int oc = static_cast<int>(i % OC_G);
```
It may be better to change the index types to `int64_t`

cyyever commented on aten/src/ATen/native/quantized/cpu/qembeddingbag_prepack.cpp:
```diff
@@ -77,7 +77,7 @@ c10::intrusive_ptr<EmbeddingPackedParamsBase> PackedEmbeddingBagWeight::prepack(
 
   for (const auto i : c10::irange(embedding_rows)) {
-    weight_bias[i] = weight_zero_points[i] * weight_scales[i] * -1;
+    weight_bias[i] = weight_zero_points.at(i) * weight_scales.at(i) * -1;
```
`at` may be slower and hurt loop optimisation (vectorisation). We rarely use `at` in intensive computing.

sampathvic commented on aten/src/ATen/native/quantized/cpu/fbgemm_utils.cpp:
```diff
@@ -531,8 +531,8 @@ int register_embedding_params() {
 
-            at::Tensor weight = std::move(tensors[0]);
-            return PackedEmbeddingBagWeight::prepack(std::move(weight));
+            const auto& weight = tensors[0];
+            return PackedEmbeddingBagWeight::prepack(weight);
```
Earlier PackedEmbeddingBagWeight::prepack was doing a copy which was unnecessary. This function is reading the object data and not doing any ownership transfer. Hence, this is a wasted copy operation.

digantdesai replied:
Can you share the details we discussed offline here for posterity?

sampathvic replied:
Sure, here is the offline discussion (removed the links as they won't work here). This function PackedEmbeddingBagWeight::prepack() is bit broken today. There are three issues:
  1) Some callers don't move the weight while calling
  2) Even if the callers move it, is of no use since the parameter is copied by value (parameter is copied since it is lacking '&&' which is needed for move semantics)
  3) This function doesn't really need move semantics as it is not taking ownership of any data:
	3.a) Gets weight_contig from the parameter and ends up copying by value when assigning to the output data
	3.b) Gets channel_scales from the parameter and ends up copying by value when creating the weight_scales vector
	3.c) Gets channel_zero_points from the parameter and ends up copying by value when creating the weight_zero_points vector

digantdesai reviewed (approved):
Thanks. Make sure the GitHub CI is green before merging.

sampathvic committed:
Addressing some linter errors (#158670)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/158670

Addressing the linter errors reported in the changed files.

Test Plan:
```
buck test mode/opt deeplearning/fbgemm:QuantUtilsTest
```
https://www.internalfb.com/intern/testinfra/testrun/11821949118528688

```
buck test mode/opt caffe2/torch/fb/model_transform/splitting/tests:split_dispatcher_test
```
https://www.internalfb.com/intern/testinfra/testrun/7881299627525465

Rollback Plan:

Reviewed By: excelle08

Differential Revision: D78352311
